packages/lab/src/overlay/OverlayPanelBase.tsx

```diff
@@ -0,0 +1,52 @@
+import { FloatingArrow } from "@floating-ui/react";
+
+import overlayCss from "./Overlay.css";
+import { useWindow } from "@salt-ds/window";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { Button, makePrefixer } from "@salt-ds/core";
+import { CloseIcon } from "@salt-ds/icons";
+import { ComponentPropsWithoutRef } from "react";
+import { useOverlayContext } from "./OverlayContext";
+import clsx from "clsx";
+
+const withBaseName = makePrefixer("saltOverlayPanelBase");
+
+// OverlayPanelBase component needed for CSS style injection
+export const OverlayPanelBase = (props: ComponentPropsWithoutRef<"div">) => {
+  const targetWindow = useWindow();
+  useComponentCssInjection({
+    testId: "salt-overlay",
+    css: overlayCss,
+    window: targetWindow,
+  });
+
+  const { arrowProps, handleCloseButton } = useOverlayContext();
+  const { children, className, ...rest } = props;
+
+  return (
+    <>
+      <div className={clsx(withBaseName("container"), className)} {...rest}>
+        <Button
+          onClick={handleCloseButton}
+          variant="secondary"
+          className={withBaseName("closeButton")}
+          aria-label="Close Overlay"
+        >
+          <CloseIcon aria-hidden />
+        </Button>
+        <div className={withBaseName("content")}>{children}</div>
+      </div>
+
+      <FloatingArrow
+        {...arrowProps}
+        strokeWidth={1}
+        fill="var(--overlay-background)"
+        stroke="var(--overlay-borderColor)"
+        style={{
```

Comment:
can these height and width styles be removed ? - i don't see a noticeable difference with or without

Reply:
these are needed for `FloatingArrows`. Without height and width, the arrows will be wonkily placed - might require a server restart to see them wonky in storybook